distributed/actor.py

```diff
@@ -176,11 +178,11 @@ async def unwrap():
                 else:
                     # TODO: this mechanism is error prone
                     # we should endeavor to make dask's standard code work here
-                    q = Queue()
+                    q = Queue(loop=self._io_loop.asyncio_loop)
```

Comment:
I recommend using `asyncio.Queue` and `import asyncio` above.  This is slightly more verbose but also keeps things explicit for future readers.

Also, we can probably drop the `loop=` keyword.  We're always in the same event loop thread.

Reply:
The queue is created outside of the async coroutine, so indeed it was picking up some other default loop, resulting in "future is on other loop" exception